lib/metababel/bt2_trace_class_generator.rb

```diff
@@ -438,14 +438,40 @@ def get_getter(field:, arg_variables:)
       bt_func_get = self.class.instance_variable_get(:@bt_func) % 'get'
       variable = bt_get_variable(arg_variables).name
       cast_func = @cast_type ? "(#{@cast_type})" : ''
-      pr "#{variable} = #{cast_func}#{bt_func_get}(#{field});"
+
+      if @cast_type and @cast_type.start_with?("struct ")
+        bt_type = self.class.instance_variable_get(:@bt_type)
+        raise "Unsupported '#{bt_type}' to '#{@cast_type}' casting." unless self.class == BTFieldClass::String
+
+        pr "// Dump string data to the struct."
+        pr "memcpy(&#{variable}, #{bt_func_get}(#{field}), sizeof(#{variable}));"
+      else
+        pr "#{variable} = #{cast_func}#{bt_func_get}(#{field});"
+      end
     end
 
     def get_setter(field:, arg_variables:)
       bt_func_get = self.class.instance_variable_get(:@bt_func) % 'set'
       variable = bt_get_variable(arg_variables).name
       cast_func = @cast_type ? "(#{self.class.instance_variable_get(:@bt_type)})" : ''
-      pr "#{bt_func_get}(#{field}, #{variable});"
+
+      if @cast_type and @cast_type.start_with?("struct ")
+        bt_type = self.class.instance_variable_get(:@bt_type) 
+        raise "Unsupported '#{@cast_type}' to '#{bt_type}' casting." unless self.class == BTFieldClass::String
+
+        pr "// Dump data to a temporal string."
+        pr "char *#{field}_temp = malloc(sizeof(pf_1));"
+        pr "assert(#{field}_temp != NULL && \"Out of memory\");"
+        pr "memcpy(#{field}_temp, &#{variable}, sizeof(#{variable}));"
+        pr "\n"
```

Comment:
No need to add 2 line
```suggestion
        pr ""
```